iOS/MobileStarter/react-app/src/HubScreen.tsx

```diff
@@ -239,247 +114,109 @@ export function HubScreen(props: HubScreenProps) {
-        openIModel(localBriefcases[0]);
-      }
-    } catch (error) {
-      // There was a problem loading the iModel. Show the error, and give the user the option
-      // to sign out.
-      presentError("LoadErrorFormat", error);
-      setButtonTitles([signOutLabel]);
-    }
-  }, [iModels, downloadIModel, openIModel, isMountedRef, signOutLabel]);
-
-  let filteredTitles = buttonTitles;
-  if (filter && hubStep === HubStep.SelectProject) {
-    filteredTitles = buttonTitles.filter((value) => value.toLocaleLowerCase().indexOf(filter) !== -1);
-  }
-  // Convert the button titles into buttons.
-  const buttons = filteredTitles.map((title: string, index: number) => {
-    const getTitle = (title: string, briefcase: LocalBriefcaseProps | undefined) => {
-      if (!briefcase) return title;
-      return title + " (" + fileSizeString(briefcase.fileSize) + ")";
-    }
-    if (hubStep === HubStep.SelectIModel || hubStep === HubStep.DownloadIModel) {
-      const briefcase = iModels[index].briefcase;
-      let deleteButton;
-      if (briefcase) {
-        deleteButton = (
-          <div className="delete-button" onClick={async (e) => {
-            e.stopPropagation();
-            try {
-              await MobileCore.deleteCachedBriefcase(briefcase);
-              if (!isMountedRef.current) return;
-              if (project) {
-                selectProject(project);
-              }
-            } catch (error) {
-              // There was a problem deleting the cached briefcase. Show the error.
-              presentError("DeleteErrorFormat", error, "HubScreen");
-            }
-          }}>
-            <IconImage iconSpec="icon-delete" />
-          </div>
-        );
-      }
-      return (
-        <div className="imodel-row" key={index}>
-          <Button
-            onClick={() => handleSelectIModel(index)}
-            title={getTitle(title, briefcase)}
-          >
-            {deleteButton}
-          </Button>
-        </div>);
-    } else {
-      return <Button
-        key={index}
-        onClick={async () => {
-          if (title === signOutLabel) {
-            if (IModelApp.authorizationClient instanceof ITMAuthorizationClient) {
+  }, [isMountedRef, project, initialized]);
+
+  let moreButton: React.ReactNode;
+  let stepContent: React.ReactNode;
+
+  switch (hubStep) {
+    case HubStep.SelectProject:
+      stepContent = <ProjectPicker signedIn={initialized}
+        onSelect={(newProject) => {
+          saveActiveProject(newProject);
+          setProject(newProject);
+          setHubStep(HubStep.SelectIModel);
+        }}
+        onError={() => setHubStep(HubStep.Error)}
+      />;
+      break;
+
+    case HubStep.SelectIModel:
+      const actions: AlertAction[] = [];
+      if (haveCachedBriefcase) {
+        actions.push({
+          name: "deleteAll",
+          title: deleteAllDownloadsLabel,
+          onSelected: async () => {
+            if (project) {
```

Comment:
It should be impossible to get to HubStep.SelectIModel while project is undefined. If that is not the case, then things should change to make it the case. If it is the case, then it should be safe to use project! anywhere a project is required.

Reply:
I just realized that this one is in a callback, so project could possibly be undefined. However, I would use the following instead of this if statement:

```ts
if (!project) return;
```

Reply:
Fixed